index.html

```diff
@@ -121,11 +125,30 @@
 var ALL_MOTE_RE = /(?:\B:\S+)|(?:\S+:\B)/g;
 var ONE_MOTE_RE = /^(?:(?::\S+:?)|(?:\S+:))$/;
 
+var img_suffixes = [
+  "jpg", "gif", "png"
+  ];
+
+function probable_picture(url) {
+  var m = /\/\S+\.([a-z0-9]+)$/.exec(url);
+  if (!m)
+    return false;
+  if (img_suffixes.indexOf(m[1]) == -1)
+    return false;
+  return true;
+}
+
 function clean(text) {
   var i = 0, m;
   text = text.replace(/&/g, '&amp;').replace(/</g, '&lt;').replace(/>/g, '&gt;');
   while((m = /\b(https?:\/\/\S+)/.exec(text.substring(i)))) {
-    var ins = '<a href="' + encodeURI(m[1]) + '">' + m[1] + '</a>';
+    var ins;
+
+    if (probable_picture(m[1])) {
+      ins = '<img class="embedded" src="' + encodeURI(m[1]) + '">';
```

Comment:
You mean wrap it with a link? I can do that. This is just passive mixed content, though.